### PR TITLE
Implement QueryClause and add small tests

### PR DESCRIPTION
**Pull request by yongpingchen**
#233 
- move `toJson`, `fromJson` methods to JsonUtilities
- add small tests

### CODE REVIEW

File: src/internal/JsonUtilities.ts [hunk 1]
```diff
@@ -390,3 +391,110 @@ export function jsonToServerCode(obj: any): ServerCode {
...
+                obj.lowerIncluded : null;
+        return new RangeClauseInQuery(field, upperLimit, upperIncluded, lowerLimit, lowerIncluded);
+    } else if (obj.type == "all") {
+        return new AllClause();
+    }
+    return null;
+}
+
+export function queryClauseToJson(clause: QueryClause): Object {
+    if (clause instanceof EqualsClauseInQuery) {
```
Comment (naoki-akimoto) [inline src/internal/JsonUtilities.ts]: There is no AllClause case. AllClause changes to `{"type":"all"}` .

File: test/small/JsonUtilitiesTest.ts [hunk 2]
```diff
@@ -723,4 +728,251 @@ describe("Test JsonUtilities for ServerCode", () => {
             })).deep.equal(new ServerCode("endpoint-1", "token", "app", { param1: "value" }));
         })
     })
+})
+
+describe('Test JsonUtilities for QueryClause', () => {
```
Comment (naoki-akimoto) [inline test/small/JsonUtilitiesTest.ts]: There is no AllClause test cases(from, to Json).

**Comment by naoki-akimoto** [comment]
@yongpingchen commented. Please confirm.

**Comment by yongpingchen** [comment]
@naoki-akimoto Fixed. Please confirm.

**Pull request merged**